### ISSUE TITLE
Refactor callbacks.R: replace file-based loss tracking with in-memory state

### ISSUE THREAD

**Issue by davidrsch** [davidrsch/lstm_re#7]
﻿## Summary
`app/logic/callbacks.R` uses a file-based approach to track training loss across epochs  it writes and reads `loss.json` files under a `directory` path for each model. This is fragile, creates side-effects outside the app's state, pollutes the filesystem in `static/`, and makes testing harder. In-memory state (reactiveValues or a closure) is the correct Rhino approach.

## Current (fragile file I/O pattern)
```r
loss_file <- paste0(directory, "loss.json")
if (!file.exists(loss_file)) {
  write_json(loss, loss_file)
} else {
  loss <- fromJSON(loss_file)
  loss <- c(loss, logs$loss)
  write_json(loss, loss_file)
}
```

## Issues
- Creates files in `static/` directories (side-effects, visible in git as untracked)  see the hashed directories currently in `lstm_re/app/static/`
- Requires `jsonlite[fromJSON, toJSON, write_json]` only for this pattern
- Not thread-safe
- Hard to clean up reliably on session end

## Proposed Approach
- Pass a mutable environment or closure to `create_callback` that accumulates loss in memory
- Or pass a `reactiveValues` slot for loss accumulation (already done for batch/epoch progress)
- Remove file I/O from `callbacks.R` entirely
- Remove the hashed cache directories from `app/static/` and add to `.gitignore`

## Related
- `app/static/231423a452cefa3a9c960415333c4196/`
- `app/static/3612903a0fbf5a28d518fa4397bd9da3/`
- `app/static/4763eee405581f8e04944d5f4bc98666/`
